docs/reported-properties.md

```diff
@@ -0,0 +1,122 @@
+# Reported properties
+
+## System Properties
+
+System properties are set one time at the beginning of the run and used to record details on the environment that the device client is running in.
+
+```json
+  {
+    "reported": {
+      "thief": {
+        "systemProperties": {
+          "language": "python",
+          "languageVersion": "3.8.2",
+          "osRelease": "#34~18.04.2-Ubuntu SMP Thu Oct "
+                                             "10 10:36:02 UTC 2019",
+          "osType": "Linux",
+          "sdkGithubBranch": null,
+          "sdkGithubCommit": null,
+          "sdkGithubRepo": null,
+          "sdkVersion": "2.4.0"
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `language` | string | Language name, one of `python`, `node`, `csharp`, `c`, or `java` |
+| `languageVersion` | string | Version of the language, such as `3.7`, `8`, etc.  Format is arbitrary and depends on language.  `|
+| `osRelease` | string | Release version of the OS, such as `Windows 10 Amd64`  Format is arbitrary as this is only for display |
+| `osType` | string | one of `Windows` or `Linux`.  Others will be added as necessary.  |
+| `sdkGithubRepro` | string | If testing against code cloned from GitHub, name of repo, such as `Azure/azure-iot-sdk-python`.  Excluded if testing against code installed from a package repository. |
+| `sdkGithubBranch` | string | If testing against code cloned from Github, name of branch, such as `master` Excluded if testing against code installed from a package repository. |
```

Comment:
Any idea how to programmatically get this (and commit)?

Reply:
No.  I was working off the assumption that we'd have 2 options for running the code:
1. running against a released SDK.  In that case, version # is sufficient for identifying what code is running.
2. running against unreleased code.  This is where repro, branch, and commit come from.  I have a few vague notions, but I've left this as an exercise for the reader.

Reply:
Good point. I'd consider this a stretch goal for now tho.